app/src/main/java/com/github/se/icebreakrr/ui/profile/ProfileEdit.kt

```diff
@@ -77,34 +77,33 @@ fun ProfileEditingScreen(
   val DESCRIPTION_MAX = 400
 
   LaunchedEffect(Unit) {
-    auth.currentUser?.let { profilesViewModel.getProfileByUid(it.uid) }
-    profilesViewModel.selectedProfile.value?.tags?.forEach { tag -> tagsViewModel.addFilter(tag) }
+    profilesViewModel.selfProfile.value?.tags?.forEach { tag -> tagsViewModel.addFilter(tag) }
   }
 
-  val isLoading = profilesViewModel.loading.collectAsState(initial = true).value
+  val isLoading = profilesViewModel.loadingSelf.collectAsState().value
   val pictureChangeState = profilesViewModel.pictureChangeState.collectAsState().value
-  val user = profilesViewModel.selfProfile.collectAsState().value!!
-  val editedCurrentProfile = profilesViewModel.editedCurrentProfile.collectAsState().value
+  val user = profilesViewModel.selfProfile.collectAsState().value
+  val editedSelfProfile = profilesViewModel.editedSelfProfile.collectAsState().value
   val tempBitmap = profilesViewModel.tempProfilePictureBitmap.collectAsState().value
 
   var catchphrase by remember {
-    mutableStateOf(TextFieldValue(editedCurrentProfile?.catchPhrase ?: user.catchPhrase))
+    mutableStateOf(TextFieldValue(editedSelfProfile?.catchPhrase ?: user!!.catchPhrase))
   }
   var description by remember {
-    mutableStateOf(TextFieldValue(editedCurrentProfile?.description ?: user.description))
+    mutableStateOf(TextFieldValue(editedSelfProfile?.description ?: user!!.description))
   }
   val expanded = remember { mutableStateOf(false) }
 
   var showDialog by remember { mutableStateOf(false) }
   var isModified by remember {
-    mutableStateOf(editedCurrentProfile != null || pictureChangeState != UNCHANGED)
+    mutableStateOf(editedSelfProfile != null || pictureChangeState != UNCHANGED)
```

Comment:
invalid: You postponed the problem by moving the `!!` as `user` is still null at some point, so the assertion will still trigger `NullPointerException`. But now the code has a lot more of `!!` and is a bit less readable

Reply:
My last push should fix this, such that user is actually never null